File: extensions/issue_id.py
def resolve_album_tag_country(ctx):
    return str(ctx["source"].get("COUNTRY", ""))

def resolve_album_tag_label(ctx):
    return str(ctx["source"].get("LABEL", ""))

def resolve_album_tag_catalognumber(ctx):
    return str(ctx["source"].get("CATALOGNUMBER", ""))

def resolve_album_tag_media(ctx):
    return str(ctx["source"].get("MEDIA", ""))

def resolve_album_tag_comment(ctx):
    val = ctx["source"].get("COMMENT")
    if val: return str(val)
    
    parts = [
        resolve_album_tag_release_year(ctx),
        resolve_album_tag_country(ctx),
        resolve_album_tag_label(ctx),
        resolve_album_tag_catalognumber(ctx)
    ]
    return " ".join([p for p in parts if p]).strip()

def resolve_album_tag_ctdbid(ctx):
    return str(ctx["source"].get("CTDBID", ""))

def resolve_album_tag_accuripid(ctx):
    return str(ctx["source"].get("ACCURIPID", ""))

def resolve_album_tag_discid(ctx):
    return str(ctx["source"].get("DISCID", ""))

File: extensions/js/issue_id.js
export function resolve_album_tag_country(ctx) {
    return String(ctx.source.COUNTRY || "");
}

export function resolve_album_tag_label(ctx) {
    return String(ctx.source.LABEL || "");
}

export function resolve_album_tag_catalognumber(ctx) {
    return String(ctx.source.CATALOGNUMBER || "");
}

export function resolve_album_tag_media(ctx) {
    return String(ctx.source.MEDIA || "");
}

export function resolve_album_tag_comment(ctx) {
    if (ctx.source.COMMENT) return String(ctx.source.COMMENT);
    
    const country = resolve_album_tag_country(ctx);
    const label = resolve_album_tag_label(ctx);
    const catNo = resolve_album_tag_catalognumber(ctx);

    if (!country && !label && !catNo) return "";

    const yyyy_mm = String(ctx.standard.RELEASE_YYYY_MM || "");
    const parts = [
        yyyy_mm.substring(0, 4),
        country,
        label,
        catNo
    ];
    return parts.filter(Boolean).join(" ").trim();
}

export function resolve_track_tag_ctdbid(ctx) {
    return String(ctx.source.CTDBID || "");
}

export function resolve_track_tag_accuripid(ctx) {
    return String(ctx.source.ACCURIPID || "");
}

export function resolve_track_tag_discid(ctx) {
    return String(ctx.source.DISCID || "");
}
